app/scripts/lib/modal_manager.js

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'jquery',
+  'velocity',
+  'velocityui'
+], function (_, $) {
+  'use strict';
+
+  function ModalManager () {
+    this._viewStates = [];
+    this.$modal = $('#modal');
+
+    // Listen for all clicks
+    $(document).on('click', _.bind(this._closeOnClick, this));
```

Comment:
Yeah good point. I think this is going away for now so we'll figure it out when we bring it back. Likely, as you suggested, we'll have an underlay element to trap clicks and not have to rely on the document.